sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/src/QuestionAnsweringClient.cs

```diff
@@ -0,0 +1,179 @@
+        /// <param name="credential">A <see cref="AzureKeyCredential"/> used to authenticate requests to the <paramref name="endpoint"/>.</param>
+        /// <param name="options">Optional <see cref="QuestionAnsweringClientOptions"/> to customize requests sent to the endpoint.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
+        public QuestionAnsweringClient(Uri endpoint, AzureKeyCredential credential, QuestionAnsweringClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            Endpoint = endpoint;
+            options ??= new QuestionAnsweringClientOptions();
+
+            Diagnostics = new ClientDiagnostics(options);
+            Pipeline = HttpPipelineBuilder.Build(
+                options,
+                new AzureKeyCredentialPolicy(credential, AuthorizationHeader));
+
+            _knowledgebaseRestClient = new(Diagnostics, Pipeline, Endpoint, options.Version);
+            _textRestClient = new(Diagnostics, Pipeline, Endpoint, options.Version);
+        }
+
+        /// <summary>
+        /// Protected constructor to allow mocking.
+        /// </summary>
+        protected QuestionAnsweringClient()
+        {
+        }
+
+        /// <summary>
+        /// Get the service endpoint for this client.
+        /// </summary>
+        public virtual Uri Endpoint { get; }
+
+        /// <summary>
+        /// Gets the <see cref="ClientDiagnostics"/> for this client.
+        /// </summary>
+        private protected virtual ClientDiagnostics Diagnostics { get; }
+
+        /// <summary>
+        /// Gets the <see cref="HttpPipeline"/> for this client.
+        /// </summary>
+        private protected virtual HttpPipeline Pipeline { get; }
+
+        /// <summary>Answers the specified question using your knowledgebase.</summary>
+        /// <param name="projectName">The name of the project to use.</param>
+        /// <param name="options">The question to answer.</param>
+        /// <param name="deploymentName">The optional deployment name of the project to use. If not specified, the production knowledgebase will be queried.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> to cancel the request.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="projectName"/> or <paramref name="options"/> is null.</exception>
+        /// <exception cref="RequestFailedException">The service returned an error. The exception contains details of the service error.</exception>
+        public virtual async Task<Response<KnowledgebaseAnswers>> QueryKnowledgebaseAsync(string projectName, KnowledgebaseQueryOptions options, string deploymentName = null, CancellationToken cancellationToken = default)
```

Comment:
@KrzysztofCwalina @tg-msft @schaabs it came up in review that we should move required parameters out of `KnowledgebaseQueryOptions` and put them as required parameters, which is either `string question` or `int qnaId`. Agreed? How does that differ from `EncryptParameters` for Key Vault, though?

Also, since not all languages support overloads, I propose adding just a `GetQuestionAnswer(string projectName, int qnaId, string deploymentName = null)`. when a `qnaId` is passed, it's the only option that matters.

/cc @annatisch @rokulka

Reply:
Also adding @johanste since this would affect Python as well, which doesn't support overloads.

Reply:
Tracking in https://github.com/Azure/azure-sdk-for-net/issues/22511